.github/workflows/ci.yml

```diff
@@ -50,6 +51,25 @@ jobs:
           DOCS_RS: 1
         with:
           command: doc
+      - name: Setup Cargo readme
+        uses: actions/cache@v2
+        with:
+          path: ~/.cargo/bin/cargo-readme
+          key: ${{ runner.os }}-cargo-readme
+      - name: Generate READMEs
+        run: |
+          cargo readme --no-title --no-indent-headings --no-license --no-badges > README.md
+          (cd sys && cargo readme --no-title --no-indent-headings --no-license --no-badges > README.md)
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
```

Comment:
@katyo Maybe this needs `if: ${{ github.event_name != 'pull_request' && !startsWith(github.ref, 'refs/tags/') }}`? Seems like it would otherwise create a PR with readme update as soon as _any_ CI build runs with documentation changes.

In https://github.com/Smithay/gbm.rs/pull/11 I'm simply failing the build if the readme doesn't match, I prefer a PR to keep the documentation and readme in sync.

Reply:
@MarijnS95 Of course, this is expected behavior.
Any such automatic created sub PRs should be merged with main PR branch before merging the last (see de17d96).
In such case we don't need editing READMEs manually at all.

Reply:
Yes, it can only create a single PR if this pushes to the same branch. But what happens if two PRs change the documentation? Will those override what the CI pushes to the readme PR? IMO it's saner to run it only on master, that way it's only regenerating things that actually changed and have been accepted. Or require contributors to update it, so that changes can be reviewed/cross-validated directly from the PR.

With the CI creating merges across the merge branch (ie. #19 wanting to merge into `re-org` here) this is only helping users to not have to install+run the command manually. I guess that's... fine?

Reply:
Updating READMEs via CI is quite useable because contributors may forget to do it manually.
I don't see serious problems in such design excepting some minor disadvantages like this.
In any case PRs needs to be rebased before merging. New update PR will be created when needed.
Even if contributor do not merged sub PR youself it still can be done later after merging main PR on master branch.